File: support_tools/swigwin-3.0.12/Examples/test-suite/go/using_extend_runme.go
package main

import . "./using_extend"

func main() {
	f := NewFooBar()
	if f.Blah(3).(int) != 3 {
		panic("blah(int)")
	}

	if f.Blah(3.5) != 3.5 {
		panic("blah(double)")
	}

	if f.Blah("hello").(string) != "hello" {
		panic("blah(char *)")
	}

	if f.Blah(3, 4).(int) != 7 {
		panic("blah(int,int)")
	}

	if f.Blah(3.5, 7.5) != (3.5 + 7.5) {
		panic("blah(double,double)")
	}

	if f.Duh(3) != 3 {
		panic("duh(int)")
	}
}

File: support_tools/swigwin-3.0.12/Examples/test-suite/octave/using_extend_runme.m
# do not dump Octave core
if exist("crash_dumps_octave_core", "builtin")
  crash_dumps_octave_core(0);
endif

using_extend

f = FooBar();
if (f.blah(3) != 3)
    error("blah(int)")
endif

if (f.blah(3.5) != 3.5)
    error("blah(double)")
endif

if (!strcmp(f.blah("hello"),"hello"))
    error("blah(char *)")
endif

if (f.blah(3,4) != 7)
    error("blah(int,int)")
endif

if (f.blah(3.5,7.5) != (3.5+7.5))
    error("blah(double,double)")
endif


if (f.duh(3) != 3)
    error("duh(int)")
endif
